code/modules/assembly/flash.dm

```diff
@@ -114,40 +114,85 @@
 /obj/item/assembly/flash/proc/flash_end()
 	set_light_on(FALSE)
 
-
+/**
+  * Handles actual flashing part of the attack
+  *
+  *	This proc is awful in every sense of the way, someone should definately refactor this whole code.
+  * Arguments:
+  * * M - Victim
+  * * user - Attacker
+  *	* power - handles the amount of confusion it gives you
+  * * targeted - determines if it was aoe or targeted
+  * * generic_message - checks if it should display default message.
+  */
 /obj/item/assembly/flash/proc/flash_carbon(mob/living/carbon/M, mob/user, power = 15, targeted = TRUE, generic_message = FALSE)
 	if(!istype(M))
 		return
 	if(user)
 		log_combat(user, M, "[targeted? "flashed(targeted)" : "flashed(AOE)"]", src)
 	else //caused by emp/remote signal
 		M.log_message("was [targeted? "flashed(targeted)" : "flashed(AOE)"]",LOG_ATTACK)
+
+
+
 	if(generic_message && M != user)
 		to_chat(M, "<span class='danger'>[src] emits a blinding light!</span>")
+
+	var/deviation = calculate_deviation(M,user)
+
+	var/datum/antagonist/rev/head/converter = user?.mind?.has_antag_datum(/datum/antagonist/rev/head)
+
+	//If you face away from someone they shouldnt notice any effects.
+	if(deviation == 2 && !converter)
+		return
+
 	if(targeted)
 		if(M.flash_act(1, 1))
 			if(M.get_confusion() < power)
 				var/diff = power * CONFUSION_STACK_MAX_MULTIPLIER - M.get_confusion()
 				M.add_confusion(min(power, diff))
-			if(user)
+			if(converter)
 				terrible_conversion_proc(M, user)
-				visible_message("<span class='danger'>[user] blinds [M] with the flash!</span>")
-				to_chat(user, "<span class='danger'>You blind [M] with the flash!</span>")
-				to_chat(M, "<span class='userdanger'>[user] blinds you with the flash!</span>")
+				if(deviation == 2)
+					to_chat(user, "<span class='notice'>You use the tacticool tier, lean over the shoulder technique to blind [M] with a flash!</span>")
+					deviation = 1
+				visible_message("<span class='danger'>[user] blinds [M] with the flash!</span>","<span class='userdanger'>[user] blinds you with the flash!</span>")
 			else
-				to_chat(M, "<span class='userdanger'>You are blinded by [src]!</span>")
-			M.Paralyze(rand(80,120))
+				to_chat(M, "<span class='userdanger'>You are flashed by [src]!</span>")
+			//easy way to make sure that you can only long stun someone who is facing in your direction
+			M.adjustStaminaLoss(rand(80,120)*(1-(deviation*0.5)))
+			M.Paralyze(rand(25,50)*(1-(deviation*0.5)))
```

Comment:
This probably needs a way to purposefully fail for the sake of traitortesting.

Reply:
only  revheads use the tacticool tier flashing, meaning that if they get cought, they will most likely be an attempt at mindshielding before sec realizes they are a revhead and kills them on the spot, this really doesnt change much.

Reply:
Perhaps we could then prevent the flash-from-behind behaviour when on help intent, so it would only be possible on any of the combat intents?

Reply:
> only revheads use the tacticool tier flashing, meaning that if they get cought, they will most likely be an attempt at mindshielding before sec realizes they are a revhead and kills them on the spot, this really doesnt change much.

you are correct, but id like an out for people who find themselves in the rare situations where it would matter.

Reply:
Probably with a feedback message about how you tried to convert them with a flash from behind, but your stance was too poor to manoeuvre and blind them.

Reply:
oh aight